app/models/website.rb

```diff
@@ -18,14 +18,30 @@ class Website < ActiveRecord::Base
 
   attr_accessor :generate
 
+  before_validation :check_origins
   before_create :generate_api_keys
   before_update :generate_api_keys, if: -> { generate }
+  before_update :check_required
   before_destroy :website_dependent
 
   def website_dependent
     website_members.each(&:delete)
   end
 
+  def check_origins( schemes: ["http://", "https://", "ftp://", "ftps://", "sftp://"] )
```

Comment:
stop setting default arguments for a method that will never be called elsewhere. Use a variable to define this default value inside the method.